game_board.py

```diff
@@ -0,0 +1,199 @@
+
+            screen.addstr(y,x,char,curses.color_pair(3))
+        elif char != "+" and char != "O" and char !="X":
+            screen.addstr(y,x,char)
+
+      
+          
+
+
+def print_choice(screen,selected_move_idx,move):
+    """Reads the move list  and prints the chars in the Screen. If selected
+
+    Keyword arguments:
+    screen -- the curses screen
+    selected_move_idx -- the currently selected row in the move
+    move -- the lists of all the "+" and their positions
+    """    
+    screen.clear()
+    print_map(screen)
+    h, w = screen.getmaxyx()
+    for idx, row in enumerate(move):
+        y = int(row[2])+ 5
+        x = int(row[1]) +  w//3      
+        if idx == selected_move_idx:
+            screen.attron(curses.color_pair(1))
+            screen.addstr(y,x, row[0])
+            screen.attroff(curses.color_pair(1))
+        else:
+            screen.addstr(y, x, row[0])
+   
+    screen.refresh()    
+
+
+def move_down(move,current_row):
+    """ Finds the "+" char which has the same xpos but an other Ypos in the move list
+    return the index of the new "+"
+
+    current_row -- the currently selected row in the move
+    move -- the lists of all the "+" and their positions
+    """
+    
+    new_row = current_row
+    
+    while True:
+        new_row += 1
+        if new_row == len(move):
+            new_row = 0
+        
+        if move[current_row][1] == move[new_row][1] and move[current_row][2] != move[new_row][2]:
+     
+          return new_row 
+
+def move_up(move,current_row):
+    """ Finds the "+" char which has the same xpos but an other Ypos in the move list
+    return the index of the new "+"
+
+    current_row -- the currently selected row in the move
+    move -- the lists of all the "+" and their positions
+    """
+    
+    new_row = current_row
+    
+    while True:
+        if new_row == 0:
+            new_row = len(move)
+        
+        new_row -= 1
+        
+        if move[current_row][1] == move[new_row][1] and move[current_row][2] !=   move[new_row][2]:
+     
+          return new_row  
+
+def stone_change(move,current_row):
+    """ Changes the "+" to a "X" of the current_row in the move list
+    returns the changed move list
+
+
+    current_row -- the currently selected row in the move
+    move -- the lists of all the "+" and their positions
+    """
+
+    move[current_row][0] = "X"
+    return move
+
+def main(screen):
+    # turn off cursor blinking
+    curses.curs_set(0)
+
+    # color scheme for selected row
+    curses.init_pair(1, curses.COLOR_BLACK, curses.COLOR_GREEN)
+
+    # specify the current selected row
+    current_row = 0
+
+    # move list
+    move = move_plus()
+     
+    
+    print_choice(screen,current_row,move)
+    while 1:
+        key = screen.getch()
+        if key == curses.KEY_LEFT and current_row > 0:  
+            current_row -= 1
+            print_choice(screen,current_row,move)
```

Comment:
A suggestion: You could put the print_choice on the first line of the while loop and then remove it from all the if cases. This would lower the amount of repeated code

Reply:
fixed